boxoffice/views/order.py

```diff
@@ -541,22 +543,17 @@ def cancel_line_item(line_item):
             doc=_(u"Ticket cancelled"), status_code=200)
 
 
-def process_partial_refund_for_order(order, form_dict):
-    form = RefundTransactionForm.from_json(form_dict, meta={'csrf': False})
-    if form.validate():
-        requested_refund_amount = form.amount.data
-        if not order.paid_amount:
-            return api_error(message='Refunds can only be issued for paid orders',
-                    status_code=403,
-                    errors=['non cancellable'])
+def process_partial_refund_for_order(data_dict):
+    order = data_dict['order']
+    form = data_dict['form']
+    request_method = data_dict['request_method']
 
-        if (order.refunded_amount + requested_refund_amount) > order.paid_amount:
-            return api_error(message='Invalid refund amount, must be lesser than net amount paid for the order',
-                    status_code=403,
-                    errors=['excess partial refund'])
+    if request_method == 'GET':
+        return jsonify(form_template=render_form(form=form, title=u"Partial refund", submit=u"Refund", with_chrome=False))
 
-        payment = OnlinePayment.query.filter_by(order=order,
-            pg_payment_status=RAZORPAY_PAYMENT_STATUS.CAPTURED).one()
+    if form.validate():
```

Comment:
As general safety principle, I'd say always use `validate_on_submit` if you're expecting a non-GET request. The previous block trapping `GET` may be moved in future code refactoring.